doc/rst/language/spec/conversions.rst

```diff
@@ -71,64 +71,62 @@ Implicit Numeric and Bool Conversions
 Implicit conversions among numeric types are allowed when all values
 representable in the source type can also be represented in the target
 type, retaining their full precision. In addition, implicit conversions
-from types ``int(64)`` and ``uint(64)`` to types ``real(64)`` and
-``complex(128)`` are allowed, even though they may result in a loss of
-precision.
+are permitted from ``int(s)`` and ``uint(s)`` values to ``real(t)`` and
+``complex(2*t)`` when ``s`` is less than or equal to ``t``, even
+though these cases may result in a loss of precision.
 
    *Rationale*.
 
-   We allow these additional conversions because they are an important
-   convenience for application programmers. Therefore we are willing to
-   lose precision in these cases. The largest real and complex types are
-   chosen to retain precision as often as as possible.
+   We allow these additional conversions because they provide an
+   important convenience for application programmers who want to mix
+   integral and floating point values in mathematical expressions, and
+   for computing using values using a specific bit-width. For these
+   benefits, the loss of precision seemed like a reasonable tradeoff,
+   particularly given that floating point types are approximate by
+   nature.
```

Comment:
Seems you should also update https://chapel-lang.org/docs/language/spec/procedures.html#determining-more-specific-functions - look for "Prefer an enum, bool, int, or uint cast to a default-sized real over another size of real or complex"

Reply:
Good catch.  I've got an edit for this that I'll push tomorrow.

Reply:
@mppf: Took all day, but I just pushed this if you could give it a look:  https://github.com/chapel-lang/chapel/pull/19749/commits/c8be00a0aa1152bd228445e7b6839fb50156b025

Reply:
looks great, thanks